### PR TITLE
Polish Kubernetes extension

### PR DESCRIPTION
**Pull request by geoand**
* Remove unused KubernetesEnvVarBuildItem
* Simplify KubernetesDeployerPrerequisite
* Use a consistent name for the logger

### CODE REVIEW

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeploy.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 public class KubernetesDeploy implements BooleanSupplier {
 
-    private final Logger LOGGER = Logger.getLogger(KubernetesDeploy.class);
+    private final Logger LOG = Logger.getLogger(KubernetesDeploy.class);
```
Comment (gastaldi) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeploy.java]:
```suggestion
    private static final Logger log = Logger.getLogger(KubernetesDeploy.class);
```

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java [hunk 2]
```diff
@@ -376,10 +371,6 @@ private void applyBuildItems(Session session,
         containerImageBuildItem
                 .ifPresent(c -> session.resources().decorate(KNATIVE, new ApplyImageDecorator(knativeName, c.getImage())));
 
-        //Handle env variables
-        kubernetesEnvBuildItems.forEach(e -> session.resources()
```
Comment (gastaldi) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java]: Interesting, so how can another extension add env variables to the k8s deployment without this snippet?

Comment (geoand) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java]: This was just a left over from a previous iteration of the s2i implementation (as confirmed by @iocanel). No extensions need to do this anymore

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeploy.java (hunk 1 above)
Comment (geoand) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeploy.java]: Good idea, done

**Review by gastaldi — APPROVED** [review]
Looks good! Good job again ;)

**Comment by geoand** [comment]
Thanks for checking!

**Pull request merged**